src/promptflow-tools/promptflow/tools/aoai_gpt4v.py

```diff
@@ -7,12 +7,130 @@
+    def __init__(
+        self,
+        name: str,
+        model_name: str,
+        version: str
+    ):
+        self.name = name
+        self.model_name = model_name
+        self.version = version
+
+
+class ListDeploymentsError(Exception):
+    def __init__(self, msg):
+        super().__init__(msg)
+
+
+def _build_deployment_dict(item) -> Deployment:
+    model = item.properties.model
+    return Deployment(item.name, model.name, model.version)
+
+
+def list_deployment_names(
+    subscription_id,
+    resource_group_name,
+    workspace_name,
+    connection: AzureOpenAIConnection = None
+) -> List[Dict[str, str]]:
+    try:
+        # Local does not support dynamic list.
+        from azure.mgmt.cognitiveservices import CognitiveServicesManagementClient
+        from promptflow.azure.operations._arm_connection_operations import ArmConnectionOperations
+    except ImportError:
+        return [{"value": ""}]
+
+    try:
+        credential = _get_credential()
+        try:
+            conn = ArmConnectionOperations._build_connection_dict(
+                name=connection,
+                subscription_id=subscription_id,
+                resource_group_name=resource_group_name,
+                workspace_name=workspace_name,
+                credential=credential
+            )
+            resource_id = conn.get("value").get('resource_id', "")
+            print(f"Connection {connection} resource id: {resource_id}")
+            conn_sub, conn_rg, conn_account = _parse_resource_id(resource_id)
+        except Exception:
+            raise Exception("Failed to get connection resource id.")
+
+        client = CognitiveServicesManagementClient(
+            credential=credential,
+            subscription_id=conn_sub,
+        )
+        deployment_collection = client.deployments.list(
+            resource_group_name=conn_rg,
+            account_name=conn_account,
+        )
+
+        res = []
+        for item in deployment_collection:
+            deployment = _build_deployment_dict(item)
+            print(f"Connection {connection} deployments:")
+            print(
+                f"deployment_name: {deployment.name}, model_name: "
+                f"{deployment.model_name}, version: {deployment.version}."
+            )
+            if deployment.version == GPT4V_VERSION:
+                print(f"{deployment.name}:{deployment.version} selected.")
+                cur_item = {
+                    "value": deployment.name,
+                    "display_value": deployment.name,
+                }
+                res.append(cur_item)
+
+        if not res:
+            message = (
+                "No deployments support gpt4v in current connection, please create "
+                "gpt4v deployments or use other connections."
+            )
+            raise ListDeploymentsError(message)
+    except ListDeploymentsError:
+        print("raise ListDeploymentsError error")
+        raise
+    except Exception as e:
```

Comment:
what's the error message looks like when user has no access to list the deployments?

Reply:
Failed to list deployments due to permission issue: (AuthorizationFailed) The client 'xxx@microsoft.com' with object id 'xxx' does not have authorization to perform action 'Microsoft.CognitiveServices/accounts/deployments/read' over scope '/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.CognitiveServices/accounts/xxx' or the scope is invalid. If access was recently granted, please refresh your credentials.
Code: AuthorizationFailed